build/makelib/golang.mk

```diff
@@ -213,12 +212,18 @@ $(CONTROLLER_GEN) $(YQ):
 		echo === installing controller-gen ;\
 		go get sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION);\
 		mv $$CONTROLLER_GEN_TMP_DIR/controller-gen $(CONTROLLER_GEN) ;\
-		echo === installing yq ;\
-		go get github.com/mikefarah/yq/v3;\
-		mv $$CONTROLLER_GEN_TMP_DIR/yq $(YQ) ;\
 		rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
 	}
 
+YQ_VERSION = v4.14.2
+YQ := $(TOOLS_HOST_DIR)/yq-$(YQ_VERSION)
+export YQ
+$(YQ):
+	@echo === installing yq $(YQ_VERSION) $(REAL_HOST_PLATFORM)
+	@mkdir -p $(TOOLS_HOST_DIR)
+	@curl -JL https://github.com/mikefarah/yq/releases/download/$(YQ_VERSION)/yq_$(REAL_HOST_PLATFORM) -o $(YQ)
+	@chmod +x $(YQ)
```

Comment:
Disentangling yq download from controller-gen download since we are changing to yq v4 for the more top-level Make targets.